### ISSUE TITLE
failed: FlowSplitWhenSpec

### ISSUE THREAD

**Issue by patriknw** [akka/akka#22097]
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/8118

```
[info] - must fail substream if materialized twice *** FAILED *** (2 seconds, 40 milliseconds)
[info]   Expected exception java.lang.IllegalStateException to be thrown, but no exception was thrown (FlowSplitWhenSpec.scala:260)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:340)
[info]   at org.scalatest.words.ResultOfAnTypeInvocation.mustBe(ResultOfAnTypeInvocation.scala:155)
[info]   at akka.stream.scaladsl.FlowSplitWhenSpec$$anonfun$2$$anonfun$apply$mcV$sp$14$$anonfun$apply$18.apply(FlowSplitWhenSpec.scala:260)
[info]   at akka.stream.scaladsl.FlowSplitWhenSpec$$anonfun$2$$anonfun$apply$mcV$sp$14$$anonfun$apply$18.apply(FlowSplitWhenSpec.scala:260)
[info]   at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:25)
[info]   at akka.stream.scaladsl.FlowSplitWhenSpec$$anonfun$2$$anonfun$apply$mcV$sp$14.apply(FlowSplitWhenSpec.scala:259)
[info]   at akka.stream.scaladsl.FlowSplitWhenSpec$$anonfun$2$$anonfun$apply$mcV$sp$14.apply(FlowSplitWhenSpec.scala:259)
```

**Comment by ktoso** [comment]
Very weird one, should look into it soon.

**Comment by drewhk** [comment]
Seems like a race condition to me:

```scala
.mapAsync(1) { src ⇒ src.runWith(Sink.ignore); src.runWith(Sink.ignore) }
```

The above snippets assumes that the second will materialize to a failed future, but in practice there might be a race so there is a chance that the second runWith is the one that actually succeeds while the first fails. Since the first Future is ignored it will not show up if it fails. At least this is my hypothesis.

**Comment by 2m** [comment]
Fixed in #22266
